fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -297,7 +298,17 @@ void deleteDocument(Tuple groupingKey, Integer partitionId, Tuple primaryKey) th
                 if (valid > 0) {
                     state.context.record(LuceneEvents.Events.LUCENE_DELETE_DOCUMENT_BY_PRIMARY_KEY, System.nanoTime() - startTime);
                     return;
+                } else if (LOG.isDebugEnabled()) {
+                    LOG.debug(KeyValueLogMessage.of("try delete document failed",
+                            LuceneLogMessageKeys.GROUP, groupingKey,
+                            LuceneLogMessageKeys.PARTITION, partitionId,
+                            LuceneLogMessageKeys.PRIMARY_KEY, primaryKey));
```

Comment:
Can we add the `segmentName` to the `DocumentIndexEntry` and include that here?

Reply:
Added.